lametro/utils.py

```diff
@@ -11,29 +12,33 @@
 
 app_timezone = pytz.timezone(settings.TIME_ZONE)
 
-# Parse the subject line from ocr_full_text
 def format_full_text(full_text):
+    '''
+    The search results and board report titles (on the BillDetail) should show the "SUBJECT:" header from the board report when present.
+    The ocr_full_text contains this information. Some example snippets:
+
+    # Subject header followed by two linebreaks.
+    ..Subject\nSUBJECT:\tFOOD SERVICE OPERATOR\n\n..Action\nACTION:\tAWARD SERVICES CONTRACT\n\n..
+
+    # Subject header followed by a return carriage and linebreak.
+    ..Subject/Action\r\nSUBJECT: MONTHLY REPORT ON CRENSHAW/LAX SAFETY\r\nACTION: RECEIVE AND FILE\r\n
+
+    # Subject header with a linebreak in the middle and without an ACTION header.
+    ..Subject\nSUBJECT:    REVISED MOTION BY DIRECTORS HAHN, SOLIS, \nGARCIA, AND DUPONT-WALKER\n..Title\n
+    '''
     results = ''
 
     if full_text:
-        txt_as_array = full_text.split("..")
-        for item in txt_as_array:
-            if 'SUBJECT:' in item:
-                array_with_subject = item.split('\n\n')
-                for item in array_with_subject:
-                    if 'SUBJECT:' in item:
-                        results = item.replace('\n', '')
+        clean_full_text = full_text.replace('\n\n', 'NEWLINE').replace('\r\n', 'NEWLINE').replace('\n..', 'NEWLINE').replace('\n', ' ')
+        match = re.search('(SUBJECT:)(.*?)(NEWLINE|ACTION:)', clean_full_text)
```

Comment:
I think moving that last replace to the line above (and seeing examples in the docstring!) really helped with the clarity here. Text parsing sucks, haha. Nice changes, all around.